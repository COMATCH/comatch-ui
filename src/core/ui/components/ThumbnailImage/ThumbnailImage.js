import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { profileImagePlaceholderBase64 } from './images/profileImagePlaceholderBase64';

import './ThumbnailImage.scss';

const propTypes = {
    alt: PropTypes.string.isRequired,
    src: PropTypes.string,
    // A list of predefined placeholder images for the thumbnail, for the time
    // being accounts only for profileImage but might grow in the future
    placeholder: PropTypes.oneOf(['', 'profileImage']),
    skipCdnUrl: PropTypes.bool,
};

const defaultProps = {
    src: '',
    placeholder: 'profileImage',
    skipCdnUrl: false,
};

function getImageSrc(src, placeholder) {
    if (src.length) {
        return src;
    }

    switch (placeholder) {
        // no placeholder
        case '':
            return '';
        // profile image placeholder
        case 'profileImage':
        default:
            return profileImagePlaceholderBase64;
    }
}

export const ThumbnailImage = ({ alt, src, placeholder }) => {
    const classes = classNames('ThumbnailImage', {
        'with-placeholder': !src.length && placeholder.length
    });
    return (
        <div className={classes}>
            <img src={getImageSrc(src, placeholder)} alt={alt} />
        </div>
    );
};

ThumbnailImage.defaultProps = defaultProps;

ThumbnailImage.propTypes = propTypes;
