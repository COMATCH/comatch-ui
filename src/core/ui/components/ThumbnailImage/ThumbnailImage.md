
## ThumbnailImage

From [`ThumbnailImage.js`](ThumbnailImage.js)



prop | type | default | required | description
---- | :----: | :-------: | :--------: | -----------
**alt** | `String` |  | :white_check_mark: | 
**placeholder** | `Enum('','profileImage')` | `'profileImage'` | :x: | 
**skipCdnUrl** | `Boolean` | `false` | :x: | 
**src** | `String` | `''` | :x: | 



