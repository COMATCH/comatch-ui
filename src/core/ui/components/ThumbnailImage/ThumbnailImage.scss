@import '../../../styles/variables.scss';

.ThumbnailImage {
    background-color: $co-color-white;
    border: 1px solid $co-color-mid-gray;
    border-radius: 3px;
    width: 170px;
    padding: 3px;

    .Image {
        width: 100%;
    }

    // States
    // - when image is uploaded and displayed
    &:not(.with-placeholder) {
        .Image {
            border: 1px solid $co-color-mid-gray;
        }
    }
}
