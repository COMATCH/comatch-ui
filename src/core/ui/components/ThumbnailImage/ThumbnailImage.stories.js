/* eslint-disable import/no-unresolved */
import React from 'react';
import Highlight from 'react-highlight';
import { storiesOf } from '@storybook/react';
import { ThumbnailImage } from './ThumbnailImage';
import { comatchLogoBase64 } from '../Spinner/images/comatchLogoBase64';

storiesOf('ThumbnailImage', module)
    .add('with profile image placeholder', () => (
        <>
            <ThumbnailImage alt="picture" />
            <Highlight className="html">{`<ThumbnailImage alt="picture" />`}</Highlight>
        </>
    ))
    .add('with no src and no placeholder', () => (
        <>
            <ThumbnailImage placeholder="" alt="picture" />
            <Highlight className="html">{`<ThumbnailImage placeholder="" alt="picture" />`}</Highlight>
        </>
    ))
    .add('with src defined', () => (
        <>
            <ThumbnailImage src={comatchLogoBase64} alt="picture" />
            <Highlight className="html">{`<ThumbnailImage src={comatchLogoBase64} alt="picture" />`}</Highlight>
        </>
    ));
