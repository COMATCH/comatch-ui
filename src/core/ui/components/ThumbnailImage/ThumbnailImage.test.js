import React from 'react';
import renderer from 'react-test-renderer';
import { ThumbnailImage } from './ThumbnailImage';
import '../../../config/tests/setup';

describe('ThumbnailImage', () => {
    it('should render with default profile picture with alt correctly', () => {
        const table = renderer.create(
            <ThumbnailImage alt="picture description" />).toJSON();

        expect(table).toMatchSnapshot();
    });

    it('should render with src picture correctly', () => {
        const table = renderer.create(
            <ThumbnailImage src="sample-picture.jpg" alt="picture description" />).toJSON();

        expect(table).toMatchSnapshot();
    });

    it('should render with no placeholder correctly', () => {
        const table = renderer.create(
            <ThumbnailImage placeholder="" alt="picture description" />).toJSON();

        expect(table).toMatchSnapshot();
    });
});
