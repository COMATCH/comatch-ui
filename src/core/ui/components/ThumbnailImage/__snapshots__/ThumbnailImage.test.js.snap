// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThumbnailImage should render with default profile picture with alt correctly 1`] = `
<div
  className="ThumbnailImage with-placeholder"
>
  <img
    alt="picture description"
    src="data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHhtbG5zOnhsaW5rPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5L3hsaW5rIiB3aWR0aD0iMTU4IiBoZWlnaHQ9IjE1NCIgdmlld0JveD0iMCAwIDE1OCAxNTQiPiAgICA8ZGVmcz4gICAgICAgIDxwYXRoIGlkPSJhIiBkPSJNMy43MzIgMzcuNzI0bC0zLjI4LTMuMzIxQTEuNTU5IDEuNTU5IDAgMCAxIDAgMzMuMzA2VjE0LjQ5NGMwLS40MTIuMTY1LS44MDguNDU3LTEuMTAyTDEzLjM0Mi40NzZhMS42IDEuNiAwIDAgMSAxLjEzMy0uNDY4aDE5LjAyMmMuNDIzIDAgLjgyOS4xNjYgMS4xMjcuNDYybDEzLjA0MyAxMi45MmMuMjk3LjI5NS40NjQuNjk0LjQ2NCAxLjExdjE4LjhjMCAuNDE0LS4xNjUuODEtLjQ1OSAxLjEwNGwtMy4zMTcgMy4zMkgzLjczMnoiLz4gICAgICAgIDxwYXRoIGlkPSJjIiBkPSJNMTAuNjMyIDQ0LjcxMUwuNDUyIDM0LjQwM0ExLjU1OSAxLjU1OSAwIDAgMSAwIDMzLjMwNlYxNC40OTRjMC0uNDEyLjE2NS0uODA4LjQ1Ny0xLjEwMkwxMy4zNDIuNDc2YTEuNiAxLjYgMCAwIDEgMS4xMzMtLjQ2OGgxOS4wMjJjLjQyMyAwIC44MjkuMTY2IDEuMTI3LjQ2MmwxMy4wNDMgMTIuOTJjLjI5Ny4yOTUuNDY0LjY5NC40NjQgMS4xMXYxOC44YzAgLjQxNC0uMTY1LjgxLS40NTkgMS4xMDRMMzcuMzczIDQ0LjcxMUgxMC42MzJ6Ii8+ICAgICAgICA8cGF0aCBpZD0iZSIgZD0iTTE0LjQ3NS4wMDhhMS42IDEuNiAwIDAgMC0xLjEzMy40NjhMLjQ1NyAxMy4zOTJjLS4yOTIuMjk0LS40NTcuNjktLjQ1NyAxLjEwMnYxOC44MTJjMCAuNDEuMTYyLjgwNC40NTIgMS4wOTdMMTMuMzQgNDcuNDU1Yy4zLjMwMy43MS40NzQgMS4xMzkuNDc0aDE5LjAxM2ExLjYgMS42IDAgMCAwIDEuMTMyLS40NjdsMTMuMDQ3LTEzLjA1OGExLjU2IDEuNTYgMCAwIDAgLjQ1OS0xLjEwM1YxNC41YzAtLjQxNi0uMTY3LS44MTUtLjQ2NC0xLjExTDM0LjYyNC40N2ExLjYwMiAxLjYwMiAwIDAgMC0xLjEyNy0uNDYySDE0LjQ3NXoiLz4gICAgICAgIDxwYXRoIGlkPSJnIiBkPSJNMTQuNDc1LjAwOGExLjYgMS42IDAgMCAwLTEuMTMzLjQ2OEwuNDU3IDEzLjM5MmMtLjI5Mi4yOTQtLjQ1Ny42OS0uNDU3IDEuMTAydjE4LjgxMmMwIC40MS4xNjIuODA0LjQ1MiAxLjA5N0wxMy4zNCA0Ny40NTVjLjMuMzAzLjcxLjQ3NCAxLjEzOS40NzRoMTkuMDEzYTEuNiAxLjYgMCAwIDAgMS4xMzItLjQ2N2wxMy4wNDctMTMuMDU4YTEuNTYgMS41NiAwIDAgMCAuNDU5LTEuMTAzVjE0LjVjMC0uNDE2LS4xNjctLjgxNS0uNDY0LTEuMTFMMzQuNjI0LjQ3YTEuNjAyIDEuNjAyIDAgMCAwLTEuMTI3LS40NjJIMTQuNDc1eiIvPiAgICAgICAgPHBhdGggaWQ9ImkiIGQ9Ik0xNC40NzUuMDA4YTEuNiAxLjYgMCAwIDAtMS4xMzMuNDY4TC40NTcgMTMuMzkyYy0uMjkyLjI5NC0uNDU3LjY5LS40NTcgMS4xMDJ2MTguODEyYzAgLjQxLjE2Mi44MDQuNDUyIDEuMDk3TDEzLjM0IDQ3LjQ1NWMuMy4zMDMuNzEuNDc0IDEuMTM5LjQ3NGgxOS4wMTNhMS42IDEuNiAwIDAgMCAxLjEzMi0uNDY3bDEzLjA0Ny0xMy4wNThhMS41NiAxLjU2IDAgMCAwIC40NTktMS4xMDNWMTQuNWMwLS40MTYtLjE2Ny0uODE1LS40NjQtMS4xMUwzNC42MjQuNDdhMS42MDIgMS42MDIgMCAwIDAtMS4xMjctLjQ2MkgxNC40NzV6Ii8+ICAgICAgICA8cGF0aCBpZD0iayIgZD0iTTE0LjQ3NS4wMDhhMS42IDEuNiAwIDAgMC0xLjEzMy40NjhMLjQ1NyAxMy4zOTJjLS4yOTIuMjk0LS40NTcuNjktLjQ1NyAxLjEwMnYxOC44MTJjMCAuNDEuMTYyLjgwNC40NTIgMS4wOTdMMTMuMzQgNDcuNDU1Yy4zLjMwMy43MS40NzQgMS4xMzkuNDc0aDE5LjAxM2ExLjYgMS42IDAgMCAwIDEuMTMyLS40NjdsMTMuMDQ3LTEzLjA1OGExLjU2IDEuNTYgMCAwIDAgLjQ1OS0xLjEwM1YxNC41YzAtLjQxNi0uMTY3LS44MTUtLjQ2NC0xLjExTDM0LjYyNC40N2ExLjYwMiAxLjYwMiAwIDAgMC0xLjEyNy0uNDYySDE0LjQ3NXoiLz4gICAgICAgIDxwYXRoIGlkPSJtIiBkPSJNMTQuNDc1LjAwOGExLjYgMS42IDAgMCAwLTEuMTMzLjQ2OEwuNDU3IDEzLjM5MmMtLjI5Mi4yOTQtLjQ1Ny42OS0uNDU3IDEuMTAydjE4LjgxMmMwIC40MS4xNjIuODA0LjQ1MiAxLjA5N0wxMy4zNCA0Ny40NTVjLjMuMzAzLjcxLjQ3NCAxLjEzOS40NzRoMTkuMDEzYTEuNiAxLjYgMCAwIDAgMS4xMzItLjQ2N2wxMy4wNDctMTMuMDU4YTEuNTYgMS41NiAwIDAgMCAuNDU5LTEuMTAzVjE0LjVjMC0uNDE2LS4xNjctLjgxNS0uNDY0LTEuMTFMMzQuNjI0LjQ3YTEuNjAyIDEuNjAyIDAgMCAwLTEuMTI3LS40NjJIMTQuNDc1eiIvPiAgICAgICAgPHBhdGggaWQ9Im8iIGQ9Ik0xNC40NzUuMDA4YTEuNiAxLjYgMCAwIDAtMS4xMzMuNDY4TC40NTcgMTMuMzkyYy0uMjkyLjI5NC0uNDU3LjY5LS40NTcgMS4xMDJ2MTguODEyYzAgLjQxLjE2Mi44MDQuNDUyIDEuMDk3TDEzLjM0IDQ3LjQ1NWMuMy4zMDMuNzEuNDc0IDEuMTM5LjQ3NGgxOS4wMTNhMS42IDEuNiAwIDAgMCAxLjEzMi0uNDY3bDEzLjA0Ny0xMy4wNThhMS41NiAxLjU2IDAgMCAwIC40NTktMS4xMDNWMTQuNWMwLS40MTYtLjE2Ny0uODE1LS40NjQtMS4xMUwzNC42MjQuNDdhMS42MDIgMS42MDIgMCAwIDAtMS4xMjctLjQ2MkgxNC40NzV6Ii8+ICAgICAgICA8cGF0aCBpZD0icSIgZD0iTTE0LjQ3NS4wMDhhMS42IDEuNiAwIDAgMC0xLjEzMy40NjhMLjQ1NyAxMy4zOTJjLS4yOTIuMjk0LS40NTcuNjktLjQ1NyAxLjEwMnYxOC44MTJjMCAuNDEuMTYyLjgwNC40NTIgMS4wOTdMMTMuMzQgNDcuNDU1Yy4zLjMwMy43MS40NzQgMS4xMzkuNDc0aDE5LjAxM2ExLjYgMS42IDAgMCAwIDEuMTMyLS40NjdsMTMuMDQ3LTEzLjA1OGExLjU2IDEuNTYgMCAwIDAgLjQ1OS0xLjEwM1YxNC41YzAtLjQxNi0uMTY3LS44MTUtLjQ2NC0xLjExTDM0LjYyNC40N2ExLjYwMiAxLjYwMiAwIDAgMC0xLjEyNy0uNDYySDE0LjQ3NXoiLz4gICAgICAgIDxwYXRoIGlkPSJzIiBkPSJNOC42NjggNS4xNjFsLTguMjEgOC4yM2MtLjI5My4yOTUtLjQ1OC42OTEtLjQ1OCAxLjEwM3YxOC44MTJjMCAuNDEuMTYyLjgwNC40NTIgMS4wOTdMMTMuMzQgNDcuNDU1Yy4zLjMwMy43MS40NzQgMS4xMzkuNDc0aDE5LjAxM2ExLjYgMS42IDAgMCAwIDEuMTMyLS40NjdsMTMuMDQ3LTEzLjA1OGExLjU2IDEuNTYgMCAwIDAgLjQ1OS0xLjEwM1YxNC41YzAtLjQxNi0uMTY3LS44MTUtLjQ2NC0xLjExTDM5LjM2IDUuMTYySDguNjY4eiIvPiAgICAgICAgPHBhdGggaWQ9InUiIGQ9Ik0xLjY5OCAxMi4xNDhsLTEuMjQgMS4yNDRjLS4yOTMuMjk0LS40NTguNjktLjQ1OCAxLjEwMnYxOC44MTJjMCAuNDEuMTYyLjgwNC40NTIgMS4wOTdMMTMuMzQgNDcuNDU1Yy4zLjMwMy43MS40NzQgMS4xMzkuNDc0aDE5LjAxM2ExLjYgMS42IDAgMCAwIDEuMTMyLS40NjdsMTMuMDQ3LTEzLjA1OGExLjU2IDEuNTYgMCAwIDAgLjQ1OS0xLjEwM1YxNC41YzAtLjQxNi0uMTY3LS44MTUtLjQ2NC0xLjExbC0xLjI1NC0xLjI0MkgxLjY5OHoiLz4gICAgICAgIDxwYXRoIGlkPSJ3IiBkPSJNMCAxOS4xMzZ2MTQuMTdjMCAuNDEuMTYyLjgwNC40NTIgMS4wOTdMMTMuMzQgNDcuNDU1Yy4zLjMwMy43MS40NzQgMS4xMzkuNDc0aDE5LjAxM2ExLjYgMS42IDAgMCAwIDEuMTMyLS40NjdsMTMuMDQ3LTEzLjA1OGExLjU2IDEuNTYgMCAwIDAgLjQ1OS0xLjEwM1YxOS4xMzZIMHoiLz4gICAgICAgIDxwYXRoIGlkPSJ5IiBkPSJNMCAyNi4xMjN2Ny4xODNjMCAuNDEuMTYyLjgwNC40NTIgMS4wOTdMMTMuMzQgNDcuNDU1Yy4zLjMwMy43MS40NzQgMS4xMzkuNDc0aDE5LjAxM2ExLjYgMS42IDAgMCAwIDEuMTMyLS40NjdsMTMuMDQ3LTEzLjA1OGExLjU2IDEuNTYgMCAwIDAgLjQ1OS0xLjEwM3YtNy4xNzhIMHoiLz4gICAgICAgIDxwYXRoIGlkPSJBIiBkPSJNMCAzMy4xMXYuMTk2YzAgLjQxLjE2Mi44MDQuNDUyIDEuMDk3TDEzLjM0IDQ3LjQ1NWMuMy4zMDMuNzEuNDc0IDEuMTM5LjQ3NGgxOS4wMTNhMS42IDEuNiAwIDAgMCAxLjEzMi0uNDY3bDEzLjA0Ny0xMy4wNThhMS41NiAxLjU2IDAgMCAwIC40NTktMS4xMDN2LS4xOTFIMHoiLz4gICAgICAgIDxwYXRoIGlkPSJDIiBkPSJNNi4wNzQgNDAuMDk3bDcuMjY3IDcuMzU4Yy4zLjMwMy43MS40NzQgMS4xMzkuNDc0aDE5LjAxM2ExLjYgMS42IDAgMCAwIDEuMTMyLS40NjdsNy4zNi03LjM2NUg2LjA3NHoiLz4gICAgICAgIDxwYXRoIGlkPSJFIiBkPSJNMTIuOTc0IDQ3LjA4NGwuMzY3LjM3Yy4zLjMwNC43MS40NzUgMS4xMzkuNDc1aDE5LjAxM2ExLjYgMS42IDAgMCAwIDEuMTMyLS40NjdsLjM3Ny0uMzc4SDEyLjk3NHoiLz4gICAgPC9kZWZzPiAgICA8ZyBmaWxsPSJub25lIiBmaWxsLXJ1bGU9ImV2ZW5vZGQiPiAgICAgICAgPHBhdGggZmlsbD0iI0VGRUZFRiIgc3Ryb2tlPSIjREREIiBkPSJNLjUuNWgxNTYuNzExdjE1M0guNXoiLz4gICAgICAgIDxwYXRoIGZpbGw9IiM3NzciIGQ9Ik02NC45NjIgODguNjQ1aDI2Ljc0NGMuNDMgMCAuODQxLS4xNjcgMS4xNDUtLjQ2NWw4LjUxLTguMzI4YTEuNjQ1IDEuNjQ1IDAgMCAxIDIuMjkxIDBsMTYuMzM3IDE1Ljk5Yy4zMDQuMjk3LjQ3NC43LjQ3NCAxLjEydjIyLjQ1M2MwIC44NzUtLjcyNSAxLjU4NS0xLjYyIDEuNTg1aC04MS4wMmMtLjg5NCAwLTEuNi0uNDY5LTEuNi0xLjM0NFY5Ni44MDJjMC0uNDIuMTUyLS42NjMuNDU1LS45NmwxNi4zMzctMTUuOTlhMS42NDUgMS42NDUgMCAwIDEgMi4yOTEgMGw4LjUxIDguMzI4Yy4zMDQuMjk4LjcxNi40NjUgMS4xNDYuNDY1IiBvcGFjaXR5PSIuNzciLz4gICAgICAgIDxnIHRyYW5zZm9ybT0idHJhbnNsYXRlKDUzLjA3MSAzMSkiPiAgICAgICAgICAgIDxtYXNrIGlkPSJiIiBmaWxsPSIjZmZmIj4gICAgICAgICAgICAgICAgPHVzZSB4bGluazpocmVmPSIjYSIvPiAgICAgICAgICAgIDwvbWFzaz4gICAgICAgICAgICA8cGF0aCBmaWxsPSIjNzc3IiBkPSJNLTIxLjY2IDM3LjcyNGwtMS4yNzItMS4yNTUgOTUuMzg1LTk0LjE5MiAxLjI3MSAxLjI1NnoiIG1hc2s9InVybCgjYikiLz4gICAgICAgICAgICA8bWFzayBpZD0iZCIgZmlsbD0iI2ZmZiI+ICAgICAgICAgICAgICAgIDx1c2UgeGxpbms6aHJlZj0iI2MiLz4gICAgICAgICAgICA8L21hc2s+ICAgICAgICAgICAgPHBhdGggZmlsbD0iIzc3NyIgZD0iTS0yMS42NiA0NC43MTFsLTEuMjcyLTEuMjU1IDk1LjM4NS05NC4xOTIgMS4yNzEgMS4yNTZ6IiBtYXNrPSJ1cmwoI2QpIi8+ICAgICAgICAgICAgPG1hc2sgaWQ9ImYiIGZpbGw9IiNmZmYiPiAgICAgICAgICAgICAgICA8dXNlIHhsaW5rOmhyZWY9IiNlIi8+ICAgICAgICAgICAgPC9tYXNrPiAgICAgICAgICAgIDxwYXRoIGZpbGw9IiM3NzciIGQ9Ik0tMjEuNjYgNTEuNjk5bC0xLjI3Mi0xLjI1Nkw3Mi40NTMtNDMuNzVsMS4yNzEgMS4yNTZ6IiBtYXNrPSJ1cmwoI2YpIi8+ICAgICAgICAgICAgPG1hc2sgaWQ9ImgiIGZpbGw9IiNmZmYiPiAgICAgICAgICAgICAgICA8dXNlIHhsaW5rOmhyZWY9IiNnIi8+ICAgICAgICAgICAgPC9tYXNrPiAgICAgICAgICAgIDxwYXRoIGZpbGw9IiM3NzciIGQ9Ik0tMjEuNjYgNTguNjg2bC0xLjI3Mi0xLjI1NiA5NS4zODUtOTQuMTkxIDEuMjcxIDEuMjU1eiIgbWFzaz0idXJsKCNoKSIvPiAgICAgICAgICAgIDxtYXNrIGlkPSJqIiBmaWxsPSIjZmZmIj4gICAgICAgICAgICAgICAgPHVzZSB4bGluazpocmVmPSIjaSIvPiAgICAgICAgICAgIDwvbWFzaz4gICAgICAgICAgICA8cGF0aCBmaWxsPSIjNzc3IiBkPSJNLTIxLjY2IDY1LjY3M2wtMS4yNzItMS4yNTYgOTUuMzg1LTk0LjE5MSAxLjI3MSAxLjI1NXoiIG1hc2s9InVybCgjaikiLz4gICAgICAgICAgICA8bWFzayBpZD0ibCIgZmlsbD0iI2ZmZiI+ICAgICAgICAgICAgICAgIDx1c2UgeGxpbms6aHJlZj0iI2siLz4gICAgICAgICAgICA8L21hc2s+ICAgICAgICAgICAgPHBhdGggZmlsbD0iIzc3NyIgZD0iTS0yMS42NiA3Mi42NmwtMS4yNzItMS4yNTYgOTUuMzg1LTk0LjE5MSAxLjI3MSAxLjI1NnoiIG1hc2s9InVybCgjbCkiLz4gICAgICAgICAgICA8bWFzayBpZD0ibiIgZmlsbD0iI2ZmZiI+ICAgICAgICAgICAgICAgIDx1c2UgeGxpbms6aHJlZj0iI20iLz4gICAgICAgICAgICA8L21hc2s+ICAgICAgICAgICAgPHBhdGggZmlsbD0iIzc3NyIgZD0iTS0yMS42NiA3OS42NDdsLTEuMjcyLTEuMjU2TDcyLjQ1My0xNS44bDEuMjcxIDEuMjU2eiIgbWFzaz0idXJsKCNuKSIvPiAgICAgICAgICAgIDxtYXNrIGlkPSJwIiBmaWxsPSIjZmZmIj4gICAgICAgICAgICAgICAgPHVzZSB4bGluazpocmVmPSIjbyIvPiAgICAgICAgICAgIDwvbWFzaz4gICAgICAgICAgICA8cGF0aCBmaWxsPSIjNzc3IiBkPSJNLTIxLjY2IDg2LjYzNGwtMS4yNzItMS4yNTZMNzIuNDUzLTguODEzbDEuMjcxIDEuMjU2eiIgbWFzaz0idXJsKCNwKSIvPiAgICAgICAgICAgIDxtYXNrIGlkPSJyIiBmaWxsPSIjZmZmIj4gICAgICAgICAgICAgICAgPHVzZSB4bGluazpocmVmPSIjcSIvPiAgICAgICAgICAgIDwvbWFzaz4gICAgICAgICAgICA8cGF0aCBmaWxsPSIjNzc3IiBkPSJNLTIxLjY2IDkzLjYyMWwtMS4yNzItMS4yNTVMNzIuNDUzLTEuODI2IDczLjcyNC0uNTd6IiBtYXNrPSJ1cmwoI3IpIi8+ICAgICAgICAgICAgPG1hc2sgaWQ9InQiIGZpbGw9IiNmZmYiPiAgICAgICAgICAgICAgICA8dXNlIHhsaW5rOmhyZWY9IiNzIi8+ICAgICAgICAgICAgPC9tYXNrPiAgICAgICAgICAgIDxwYXRoIGZpbGw9IiM3NzciIGQ9Ik0tMjEuNjYgMTAwLjYwOGwtMS4yNzItMS4yNTVMNzIuNDUzIDUuMTZsMS4yNzEgMS4yNTZ6IiBtYXNrPSJ1cmwoI3QpIi8+ICAgICAgICAgICAgPG1hc2sgaWQ9InYiIGZpbGw9IiNmZmYiPiAgICAgICAgICAgICAgICA8dXNlIHhsaW5rOmhyZWY9IiN1Ii8+ICAgICAgICAgICAgPC9tYXNrPiAgICAgICAgICAgIDxwYXRoIGZpbGw9IiM3NzciIGQ9Ik0tMjEuNjYgMTA3LjU5NmwtMS4yNzItMS4yNTYgOTUuMzg1LTk0LjE5MiAxLjI3MSAxLjI1NnoiIG1hc2s9InVybCgjdikiLz4gICAgICAgICAgICA8bWFzayBpZD0ieCIgZmlsbD0iI2ZmZiI+ICAgICAgICAgICAgICAgIDx1c2UgeGxpbms6aHJlZj0iI3ciLz4gICAgICAgICAgICA8L21hc2s+ICAgICAgICAgICAgPHBhdGggZmlsbD0iIzc3NyIgZD0iTS0yMS42NiAxMTQuNTgzbC0xLjI3Mi0xLjI1NiA5NS4zODUtOTQuMTkxIDEuMjcxIDEuMjU1eiIgbWFzaz0idXJsKCN4KSIvPiAgICAgICAgICAgIDxtYXNrIGlkPSJ6IiBmaWxsPSIjZmZmIj4gICAgICAgICAgICAgICAgPHVzZSB4bGluazpocmVmPSIjeSIvPiAgICAgICAgICAgIDwvbWFzaz4gICAgICAgICAgICA8cGF0aCBmaWxsPSIjNzc3IiBkPSJNLTIxLjY2IDEyMS41N2wtMS4yNzItMS4yNTYgOTUuMzg1LTk0LjE5MSAxLjI3MSAxLjI1NXoiIG1hc2s9InVybCgjeikiLz4gICAgICAgICAgICA8bWFzayBpZD0iQiIgZmlsbD0iI2ZmZiI+ICAgICAgICAgICAgICAgIDx1c2UgeGxpbms6aHJlZj0iI0EiLz4gICAgICAgICAgICA8L21hc2s+ICAgICAgICAgICAgPHBhdGggZmlsbD0iIzc3NyIgZD0iTS0yMS42NiAxMjguNTU3bC0xLjI3Mi0xLjI1Nkw3Mi40NTMgMzMuMTFsMS4yNzEgMS4yNTV6IiBtYXNrPSJ1cmwoI0IpIi8+ICAgICAgICAgICAgPG1hc2sgaWQ9IkQiIGZpbGw9IiNmZmYiPiAgICAgICAgICAgICAgICA8dXNlIHhsaW5rOmhyZWY9IiNDIi8+ICAgICAgICAgICAgPC9tYXNrPiAgICAgICAgICAgIDxwYXRoIGZpbGw9IiM3NzciIGQ9Ik0tMjEuNjYgMTM1LjU0NGwtMS4yNzItMS4yNTYgOTUuMzg1LTk0LjE5MSAxLjI3MSAxLjI1NXoiIG1hc2s9InVybCgjRCkiLz4gICAgICAgICAgICA8bWFzayBpZD0iRiIgZmlsbD0iI2ZmZiI+ICAgICAgICAgICAgICAgIDx1c2UgeGxpbms6aHJlZj0iI0UiLz4gICAgICAgICAgICA8L21hc2s+ICAgICAgICAgICAgPHBhdGggZmlsbD0iIzc3NyIgZD0iTS0yMS42NiAxNDIuNTMxbC0xLjI3Mi0xLjI1NiA5NS4zODUtOTQuMTkxIDEuMjcxIDEuMjU2ek0tMjEuNjYgMTQ5LjUxOGwtMS4yNzItMS4yNTVMNzIuNDUzIDU0LjA3bDEuMjcxIDEuMjU2ek0tMjEuNjYgMTU2LjUwNWwtMS4yNzItMS4yNTUgOTUuMzg1LTk0LjE5MiAxLjI3MSAxLjI1NnpNLTIxLjY2IDE2My40OTJsLTEuMjcyLTEuMjU1IDk1LjM4NS05NC4xOTIgMS4yNzEgMS4yNTZ6TS0yMS42NiAxNzAuNDhsLTEuMjcyLTEuMjU2IDk1LjM4NS05NC4xOTIgMS4yNzEgMS4yNTZ6IiBtYXNrPSJ1cmwoI0YpIi8+ICAgICAgICA8L2c+ICAgIDwvZz48L3N2Zz4="
  />
</div>
`;

exports[`ThumbnailImage should render with no placeholder correctly 1`] = `
<div
  className="ThumbnailImage"
>
  <img
    alt="picture description"
    src=""
  />
</div>
`;

exports[`ThumbnailImage should render with src picture correctly 1`] = `
<div
  className="ThumbnailImage"
>
  <img
    alt="picture description"
    src="sample-picture.jpg"
  />
</div>
`;
